google-http-client-apache-v5/src/test/java/com/google/api/client/http/apache/v5/Apache5HttpTransportTest.java

```diff
@@ -0,0 +1,427 @@
+    }
+    assumeTrue(credentials != null);
+    GsonFactory jsonFactory = GsonFactory.getDefaultInstance();
+    CloudResourceManager.Builder resourceManagerBuilder =
+        new CloudResourceManager.Builder(
+                transport, jsonFactory, new HttpCredentialsAdapter(credentials))
+            .setApplicationName("Example Java App");
+    CloudResourceManager cloudResourceManager = resourceManagerBuilder.build();
+
+    Projects projects = cloudResourceManager.projects();
+    Get get = projects.get("projects/" + PROJECT_ID);
+    Project project = get.execute();
+    System.out.println("Project display name: " + project.getDisplayName());
+
+    // temp test to confirm re-usage of client works with several consecutive requests
+    projects.get("projects/" + PROJECT_ID).execute();
+    projects.get("projects/" + PROJECT_ID).execute();
+    projects.get("projects/" + PROJECT_ID).execute();
+    projects.get("projects/" + PROJECT_ID).execute();
+  }
+
+  @Test
+  public void testClientUsingDefaultApacheV5Transport() throws IOException {
+    HttpTransport transport = new Apache5HttpTransport();
+    testRequest(transport);
+  }
+
+  @Test
+  public void testClientUsingApacheV5TransportWithCustomHttpClient() throws IOException {
+    CloseableHttpClient client =
+        HttpClients.custom().disableAutomaticRetries().disableRedirectHandling().build();
+    HttpTransport transport = new Apache5HttpTransport(client);
+    testRequest(transport);
+  }
+
+  @Test
+  public void testApacheHttpTransport() {
+    Apache5HttpTransport transport = new Apache5HttpTransport();
+    checkHttpTransport(transport);
+    assertFalse(transport.isMtls());
+  }
+
+  @Test
+  public void testApacheHttpTransportWithParam() {
+    Apache5HttpTransport transport = new Apache5HttpTransport(HttpClients.custom().build(), true);
+    checkHttpTransport(transport);
+    assertTrue(transport.isMtls());
+  }
+
+  @Test
+  public void testNewDefaultHttpClient() {
+    HttpClient client = Apache5HttpTransport.newDefaultHttpClient();
+    checkHttpClient(client);
+  }
+
+  private void checkHttpTransport(Apache5HttpTransport transport) {
+    assertNotNull(transport);
+    HttpClient client = transport.getHttpClient();
+    checkHttpClient(client);
+  }
+
+  private void checkHttpClient(HttpClient client) {
+    assertNotNull(client);
+    // TODO(chingor): Is it possible to test this effectively? The newer HttpClient implementations
+    // are read-only and we're testing that we built the client with the right configuration
+  }
+
+  @Test
+  public void testRequestsWithContent() throws IOException {
+    // This test ensures that requests that have content are rejected, as opposed to those with an
+    // entity set.
+    // It is currently failing because we don't perform this check in Apache5HttpRequest.execute();
+    // TODO(diegomarquezp): find out whether this check is necessary in Apache 5.
```

Comment:
This test was meant to confirm this piece of the old client works:

https://github.com/googleapis/google-http-java-client/blob/094dcc87f22d16686242ce6cb06151b2a199ec2a/google-http-client-apache-v2/src/main/java/com/google/api/client/http/apache/v2/ApacheHttpRequest.java#L60-L63

We need to confirm if we still need this check

Reply:
Confirmed via tests that we don't need this check. Also wrote details in internal doc